packages/react-router-dom/index.tsx

```diff
@@ -1042,6 +1041,14 @@ function useScrollRestoration({
   let matches = useMatches();
   let navigation = useNavigation();
 
+  const getScrollElement = () => (id && document.getElementById(id)) || window;
+
+  // window has scrollY but normal elements have scrollTop
+  const getScrollY = () => {
+    const el = getScrollElement();
+    return el ? el.scrollY || el.scrollTop : undefined;
```

Comment:
This doesn't typecheck. I'm sure Ryan wanted to do this differently anyway, this was just how I did it in my patch.